### DISCUSSION TITLE
Why my plugin didn't works when I access directly to a specific path ? (SSR true)

### DISCUSSION THREAD

**Discussion by edusoapboxx** [nuxt/nuxt#17904] [Questions] [answered]
![Screen Shot 2022-02-24 at 13 38 49](https://user-images.githubusercontent.com/65977538/155567767-5c6e88b2-1458-45f3-9820-7d27f4b914df.png)
I'm created a plugin to use globally, and also I created a page called demo. The plugin works but give me 500 when I access directly to the url /demo or refreshing the browser into this section.

**This is the plug:**

`import { defineNuxtPlugin } from '#app'
import { createDeviceDetector } from "next-vue-device-detector";

export default defineNuxtPlugin(() => {
    const device = createDeviceDetector();
    return {
        provide: {
            isMobile: () => device.mobile
        }
    }
})`

I'm using the latest version of nuxt3

**Comment by Aareksio** [comment, marked as answer]
`next-vue-device-detector` is a client only library ([src](https://github.com/dreambo8563/next-vue-device-detector/blob/main/src/lib/main.ts#L5)). You must use it as client-only plugin. This is something you already seem to do.

The problem with client-side plugins - they are available on client side only. You must NOT access them during SSR. Either protect the use with `isClient` or make sure you never access `isMobile` before `mounted` event.

**Reply by edusoapboxx** [reply]
Thanks a lot @Aareksio  I used <ClientOnly> to protect it and works !. I will investigate a little more about other options for it.
